util/types/mytime.go

```diff
@@ -112,6 +112,63 @@ func newMysqlTime(year, month, day, hour, minute, second, microsecond int) mysql
 	}
 }
 
+func calcTimeFromSec(to *mysqlTime, seconds, microseconds int) {
+	to.hour = uint8(seconds / 3600)
+	seconds = seconds % 3600
+	to.minute = uint8(seconds / 60)
+	to.second = uint8(seconds % 60)
+	to.microsecond = uint32(microseconds)
+}
+
+// calcTimeDiff calculates difference between two datetime values as seconds + microseconds.
+// t1 and t2 should be TIME/DATE/DATETIME value.
+// sign can be +1 or -1, and t2 is preprocessed with sign first.
+func calcTimeDiff(t1, t2 TimeInternal, sign int) (seconds, microseconds int, neg bool) {
```

Comment:
Nothing confusing here: if a time is stored in microseconds format, it may overflow one int, so separate into two ints. 
Go also use this trick also, it store seconds and nanoseconds separately...

By the way, there are many feature in mysql time really make me confused, but TiDB should still compatible to it.